fs/fs.go

```diff
@@ -456,19 +473,78 @@ func (self *RealFsInfo) GetDirUsage(dir string, timeout time.Duration) (uint64,
 	err = cmd.Wait()
 	timer.Stop()
 	if err != nil {
-		return 0, fmt.Errorf("du command failed on %s with output stdout: %s, stderr: %s - %v", dir, string(stdoutb), string(stderrb), err)
+		return 0, fmt.Errorf("cmd %v failed with output stdout: %s, stderr: %s - %v", cmd.Args, string(stdoutb), string(stderrb), err)
 	}
 	stdout := string(stdoutb)
 	if souterr != nil {
 		glog.Errorf("failed to read from stdout for cmd %v - %v", cmd.Args, souterr)
 	}
 	usageInKb, err := strconv.ParseUint(strings.Fields(stdout)[0], 10, 64)
 	if err != nil {
-		return 0, fmt.Errorf("cannot parse 'du' output %s - %s", stdout, err)
+		return 0, fmt.Errorf("cannot parse cmd %v output %s - %s", cmd.Args, stdout, err)
 	}
 	return usageInKb * 1024, nil
 }
 
+func (self *RealFsInfo) GetDirInodeUsage(dir string, timeout time.Duration) (uint64, error) {
+	if dir == "" {
+		return 0, fmt.Errorf("invalid directory")
+	}
+	claimFindToken()
+	defer releaseFindToken()
+	findCmd := exec.Command("find", dir, "-xdev")
+	wcCmd := exec.Command("wc", "-l")
+	wcCmd.Stdin, _ = findCmd.StdoutPipe()
+
+	stdoutp, err := wcCmd.StdoutPipe()
+	if err != nil {
+		return 0, fmt.Errorf("failed to setup stdout for cmd %v - %v", wcCmd.Args, err)
+	}
+	findStderrp, err := findCmd.StderrPipe()
+	if err != nil {
+		return 0, fmt.Errorf("failed to setup stderr for cmd %v - %v", findCmd.Args, err)
+	}
+	wcStderrp, err := wcCmd.StderrPipe()
+	if err != nil {
+		return 0, fmt.Errorf("failed to setup stderr for cmd %v - %v", wcCmd.Args, err)
+	}
+
+	if err := findCmd.Start(); err != nil {
+		return 0, fmt.Errorf("failed to exec cmd %v - %v", findCmd.Args, err)
+	}
+
+	if err := wcCmd.Start(); err != nil {
+		return 0, fmt.Errorf("failed to exec cmd %v - %v", wcCmd.Args, err)
+	}
+	stdoutb, souterr := ioutil.ReadAll(stdoutp)
+	findStderrb, _ := ioutil.ReadAll(findStderrp)
+	wcStderrb, _ := ioutil.ReadAll(wcStderrp)
+	timer := time.AfterFunc(timeout, func() {
+		glog.Infof("killing cmd %v, and cmd %v due to timeout(%s)", findCmd.Args, wcCmd.Args, timeout.String())
+		wcCmd.Process.Kill()
+		findCmd.Process.Kill()
```

Comment:
A timeout should be an error. Does killing the findCmd cause `Wait()` to return an error?